src/js/filters/filters/postsFilter.mjs

```diff
@@ -0,0 +1,23 @@
+export function postsFilter(post, searchValue) {
+  if (post.body) {
+    if (
+      post.body.toLowerCase().includes(searchValue.toLowerCase()) ||
+      post.title.toLowerCase().includes(searchValue.toLowerCase()) ||
+      post.author.name.toLowerCase().includes(searchValue.toLowerCase())
+    )
+      return post;
+  } else {
+    if (
```

Comment:
Yes it is repetitive and I would like to refactor. As you see, my problem is that not all the posts have body. 

I am afraid I don't quite get the second part "const matches = " in your suggestion.
I want to filter out and return a list of posts which contains the searched term. Could you break it down a little bit so that I can understand?

Reply:
Matches could be renamed `const someOfTheThingsToCheckMatch`, it will either be true or false

Reply:
You could set a default value in the constructor:

```js
export class Example {
  constructor(title = "", body = "", media = "", etc = "") {}
}
```

Reply:
Oh no this is unreadable...how do you paste a code in a comment 
with markdown...?

Reply:
```js
const example = 123;
```

\`\`\`js
const example = 123;
\`\`\`

<img width="168" alt="image" src="https://user-images.githubusercontent.com/10801540/206395901-919f7a37-dc55-4857-b85a-f946180fd0ad.png">